data/lua/wml/random_placement.lua

```diff
@@ -14,7 +14,7 @@ wesnoth.wml_actions.random_placement = function(cfg)
 	local math_abs = math.abs	
 	local locs = wesnoth.get_locations(filter)
 	if type(num_items) == "string" then		
-		num_items = math.floor(loadstring("local size = " .. #locs .. "; return " .. num_items)())
+		num_items = math.floor(load("local size = " .. #locs .. "; return " .. num_items)())
```

Comment:
And now you know why I said this looked like the sort of thing which keeps web site admins up at night.

The string "num_items" is given the number of items as a local variable, should it need it.

Luckily,the os library is not available. So, which this can wreck havoc on a game, it should not result in a break-out.